### PR TITLE
Adds a customizable TextView component

### PR DESCRIPTION
**Pull request by manivoxel51**
## What changes are proposed in this pull request?

@imanjra @Br2850 lmk if this is on the right track for the use cases

## How is this patch tested? If it is not, please explain why.

Using this snippet 
```
def render(self, ctx):
        panel = types.Object()

        text_view_0 = types.TextView(
            title="Voxel",
            variant="body1",
            fontSize="1.2rem",
            color="primary",
            bold=True,
            textTransform="uppercase",
            width="auto",
            displayMode="inline-block",
            padding={0}
        )
        text_view_01 = types.TextView(
            title="Fiftyone",
            variant="body1",
            fontSize="1.2rem",
            color="red",
            bold=False,
            italic=True,
            textTransform="uppercase",
            width="auto",
            displayMode="inline-block",
            padding={0}
        )

        text_view_1 = types.TextView(
            title="Primary Bold Text",
            variant="body1",
            fontSize="1.2rem",
            color="primary",
            bold=True,
            textTransform="uppercase",
            align="center",
            width="100%",
            displayMode="block"
        )

        text_view_2 = types.TextView(
            title="Italic Secondary Text",
            variant="body2",
            fontSize="1rem",
            color="secondary",
            italic=True,
            align="left",
            noWrap=True,
            textDecoration="underline",
            letterSpacing="0.1em",
            displayMode="inline"
        )

        text_view_3 = types.TextView(
            title="Error Message",
            variant="caption",
            color="error",
            bold=True,
            textTransform="capitalize",
            align="right",
            fontFamily="monospace",
            letterSpacing="0.2em",
            lineHeight="1.5",
            displayMode="block"
        )

        text_view_4 = types.TextView(
            title="Success Notification",
            variant="h6",
            color="success",
            fontSize="1.5rem",
            textTransform="uppercase",
            bold=True,
            italic=True,
            align="center",
            textDecoration="none",
            displayMode="inline"
        )

        text_view_5 = types.TextView(
            title="Warning Alert",
            variant="subtitle1",
            color="warning",
            fontSize="1.25rem",
            align="left",
            letterSpacing="0.05em",
            bold=True,
            lineHeight="1.2",
            displayMode="block"
        )

        text_view_6 = types.TextView(
            title="Info Banner",
            variant="h5",
            color="info",
            fontSize="2rem",
            bold=True,
            textTransform="uppercase",
            noWrap=True,
            align="center",
            fontFamily="serif",
            letterSpacing="0.15em",
            displayMode="block"
        )

        text_view_7 = types.TextView(
            title="Disabled Text",
            variant="body1",
            color="textSecondary",
            fontSize="0.9rem",
            italic=True,
            align="right",
            textDecoration="line-through",
            lineHeight="1.6",
            displayMode="inline"
        )

        text_view_8 = types.TextView(
            title="Subtitle Example",
            variant="subtitle2",
            color="primary",
            fontSize="1.1rem",
            bold=True,
            italic=False,
            textTransform="capitalize",
            align="left",
            noWrap=False,
            displayMode="block"
        )

        text_view_9 = types.TextView(
            title="Small Caption",
            variant="caption",
            color="textSecondary",
            fontSize="0.75rem",
            italic=True,
            textTransform="none",
            align="center",
            letterSpacing="0.05em",
            lineHeight="1.4",
            displayMode="inline"
        )

        text_view_10 = types.TextView(
            title="Large Headline",
            variant="h4",
            fontSize="2.5rem",
            color="primary",
            bold=True,
            align="center",
            textTransform="uppercase",
            letterSpacing="0.3em",
            lineHeight="1.1",
            displayMode="block"
        )

        text_view_11 = types.TextView(
            title="Quote",
            variant="body1",
            color="secondary",
            fontStyle="italic",
            fontSize="1.1rem",
            align="left",
            letterSpacing="normal",
            lineHeight="1.7",
            displayMode="block"
        )

        text_view_12 = types.TextView(
            title="Highlighted Text",
            variant="h6",
            fontSize="1.8rem",
            color="error",
            bold=True,
            textDecoration="underline",
            textTransform="capitalize",
            align="right",
            noWrap=False,
            displayMode="block"
        )

        text_view_13 = types.TextView(
            title="Lowercase Text",
            variant="body2",
            color="textPrimary",
            textTransform="lowercase",
            fontSize="1rem",
            align="left",
            letterSpacing="normal",
            lineHeight="1.3",
            displayMode="inline"
        )

        text_view_14 = types.TextView(
            title="Uppercase Title",
            variant="h5",
            color="info",
            textTransform="uppercase",
            fontSize="2rem",
            bold=True,
            italic=True,
            align="center",
            letterSpacing="0.2em",
            displayMode="block"
        )

        text_view_15 = types.TextView(
            title="Strikethrough Text",
            variant="caption",
            fontSize="0.9rem",
            color="error",
            textDecoration="line-through",
            italic=True,
            align="right",
            letterSpacing="0.05em",
            displayMode="inline"
        )

        text_view_16 = types.TextView(
            title="Outlined Subtitle",
            variant="subtitle1",
            color="primary",
            bold=True,
            fontSize="1.3rem",
            textTransform="none",
            align="left",
            noWrap=False,
            letterSpacing="0.1em",
            displayMode="block"
        )

        text_view_17 = types.TextView(
            title="Center Aligned Text",
            variant="h6",
            fontSize="1.6rem",
            color="textSecondary",
            align="center",
            bold=True,
            italic=False,
            letterSpacing="normal",
            lineHeight="1.4",
            displayMode="block"
        )

        text_view_18 = types.TextView(
            title="Blockquote",
            variant="body1",
            color="textPrimary",
            fontSize="1.2rem",
            italic=True,
            align="left",
            lineHeight="1.8",
            letterSpacing="0.1em",
            displayMode="inline"
        )

        text_view_19 = types.TextView(
            title="Notification Title",
            variant="h5",
            fontSize="2rem",
            color="success",
            bold=True,
            textTransform="capitalize",
            align="center",
            noWrap=True,
            lineHeight="1.5",
            displayMode="block"
        )

        text_view_20 = types.TextView(
            title="Overline Text",
            variant="overline",
            fontSize="0.8rem",
            color="warning",
            textTransform="uppercase",
            align="right",
            letterSpacing="0.3em",
            displayMode="inline"
        )

        text_view_21 = types.TextView(
            title="Info Box",
            variant="body2",
            fontSize="1rem",
            color="info",
            bold=True,
            align="center",
            lineHeight="1.3",
            displayMode="block"
        )

        text_view_22 = types.TextView(
            title="Secondary Block Text",
            variant="body1",
            fontSize="1rem",
            color="secondary",
            italic=True,
            align="left",
            noWrap=False,
            textTransform="none",
            displayMode="block"
        )

        text_view_23 = types.TextView(
            title="Primary Subtitle",
            variant="subtitle1",
            fontSize="1.2rem",
            color="primary",
            bold=True,
            align="left",
            noWrap=True,
            letterSpacing="normal",
            displayMode="inline"
        )

        text_view_24 = types.TextView(
            title="Emphasized Error Text",
            variant="h6",
            fontSize="1.6rem",
            color="error",
            bold=True,
            italic=True,
            align="center",
            textTransform="uppercase",
            displayMode="block"
        )

        text_view_25 = types.TextView(
            title="Underlined Caption",
            variant="caption",
            fontSize="0.8rem",
            color="textPrimary",
            textDecoration="underline",
            align="left",
            letterSpacing="0.05em",
            displayMode="inline"
        )

        panel.view("text_view_0", text_view_0)
        panel.view("text_view_01", text_view_01)

        panel.view("text_view_1", text_view_1)
        panel.view("text_view_2", text_view_2)
        panel.view("text_view_3", text_view_3)
        panel.view("text_view_4", text_view_4)
        panel.view("text_view_5", text_view_5)
        panel.view("text_view_6", text_view_6)
        panel.view("text_view_7", text_view_7)
        panel.view("text_view_8", text_view_8)
        panel.view("text_view_9", text_view_9)
        panel.view("text_view_10", text_view_10)
        panel.view("text_view_11", text_view_11)
        panel.view("text_view_12", text_view_12)
        panel.view("text_view_13", text_view_13)
        panel.view("text_view_14", text_view_14)
        panel.view("text_view_15", text_view_15)
        panel.view("text_view_16", text_view_16)
        panel.view("text_view_17", text_view_17)
        panel.view("text_view_18", text_view_18)
        panel.view("text_view_19", text_view_19)
        panel.view("text_view_20", text_view_20)
        panel.view("text_view_21", text_view_21)
        panel.view("text_view_22", text_view_22)
        panel.view("text_view_23", text_view_23)
        panel.view("text_view_24", text_view_24)
        panel.view("text_view_25", text_view_25)


        panel_view = types.GridView(
            width=100, height=100, align_x="center", align_y="center"
        )
        return types.Property(panel, view=panel_view)
```

https://github.com/user-attachments/assets/d962260e-7b6b-4c0b-8d68-b15df2c1f687

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `TextView` component for rendering customizable text based on schema properties.
	- Added a `TextView` class to enhance text view representation in the operator framework.
	- Expanded functionality with a new `TextView` class for displaying text in the FiftyOne framework.

- **Documentation**
	- Updated exports to include the new `TextView` component in the relevant modules.

### CODE REVIEW

**Review by lanzhenw — APPROVED** [review]
🚀

File: app/packages/core/src/plugins/SchemaIO/components/TextView.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    color = "primary",
+    fontSize = "1rem",
+    title = "",
+    textTransform = "none",
+    variant = "body1",
+    bold = false,
+    italic = false,
+    align = "inherit",
+    noWrap = false,
+    textDecoration = "none",
```
Comment (imanjra) [inline app/packages/core/src/plugins/SchemaIO/components/TextView.tsx:19]: @manivoxel51 Sorry for the late review. We should support snake case for everything that is set from Python side

**Review by imanjra — COMMENTED** [review]
Nice work! 🚀

File: app/packages/core/src/plugins/SchemaIO/components/TextView.tsx (hunk 1 above)
Comment (manivoxel51) [reply app/packages/core/src/plugins/SchemaIO/components/TextView.tsx:19]: you are right! will patch it shortly 🙇

**Pull request merged**